packages/NodeTypeResolver/TypeAnalyzer/MethodTypeAnalyzer.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\NodeTypeResolver\TypeAnalyzer;
+
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Identifier;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\TypeWithClassName;
+use Rector\NodeTypeResolver\NodeTypeResolver;
+
+final class MethodTypeAnalyzer
+{
+    public function __construct(
+        private NodeTypeResolver $nodeTypeResolver
+    ) {
+    }
+
+    /**
+     * @param class-string $expectedClass
+     * @param non-empty-string $expectedMethod
+     */
+    public function isCallTo(MethodCall $methodCall, string $expectedClass, string $expectedMethod): bool
+    {
+        if (! $this->isMethodName($methodCall, $expectedMethod)) {
+            return false;
+        }
+
+        return $this->isInstanceOf($methodCall->var, $expectedClass);
+    }
+
+    /**
+     * @param non-empty-string $expectedName
+     */
+    private function isMethodName(MethodCall $methodCall, string $expectedName): bool
+    {
+        if ($methodCall->name instanceof Identifier) {
+            $comparison = strcasecmp($methodCall->name->toString(), $expectedName);
```

Comment:
this may be needed to be moved to separate private method to be reusable so can change in one place when there is logic change for comparison.

Reply:
Done